pkg/cache/internal/informers.go

```diff
@@ -300,6 +339,21 @@ func (ip *Informers) Get(ctx context.Context, gvk schema.GroupVersionKind, obj r
 	return started, i, nil
 }
 
+// Remove removes an informer entry and stops it if it was running.
+func (ip *Informers) Remove(gvk schema.GroupVersionKind, obj runtime.Object) {
```

Comment:
Who  can use this func?

Reply:
It's intended to be used by whatever is responsible for knowing when an informer is no longer needed.

In the short term, this responsibility is owned by individual users. Longer-term, it sounds like controller-runtime wants to have some kind of solution for this (reference counting has been floated).

I think it's useful to have this function public as:

1) Otherwise, users of controller-runtime will be blocked from using dynamic informers until a solution is agreed on/implemented (original issue, #1884, is over a year old without a definitive path forward, so this is definitely not a trivial task). Without this, they have no way of cleaning up unused state.

2) Having this method public will provide an escape hatch should controller-runtime's chosen model of managing informers be incompatible with a given project.